### PR TITLE
Databricks App Submission

### PR DESCRIPTION
**Pull request by rapoth**
This PR captures more detailed instructions for submitting apps to Databricks.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [ ] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ ] You have included any necessary tests in the same PR.

### CODE REVIEW

File: deployment/README.md [hunk 1]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
+  5. Upload the files you downloaded and modified to your Databricks cluster
+     ```
+     cd <path-to-init-db-and-install-worker>
+     databricks fs cp init-db.sh dbfs:/spark-dotnet/
+     databricks fs cp install-worker.sh dbfs:/spark-dotnet/
+     ```
+  6. Goto to your Databricks cluster homepage -> Clusters (on the left-side menu) -> Create Cluster
+  7. After configuring the cluster appropriately, set the init script (see the image below) and create the cluster.
+     
+     <img src="../docs/img/deployment-databricks-init-script.PNG" alt="ScriptActionImage" width="500"/>
```
Comment (imback82) [inline deployment/README.md]: this image looks blurred, try different width?

File: deployment/README.md [hunk 2]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
+  2. Modify **db-init.sh** appropriately to point to the Microsoft.Spark.Worker release you want to download and install on your cluster
+  3. Download and install [Databricks CLI](https://docs.databricks.com/user-guide/dev-tools/databricks-cli.html)
+  4. [Setup authentication](https://docs.databricks.com/user-guide/dev-tools/databricks-cli.html#set-up-authentication) details for the Databricks CLI appropriately
+  5. Upload the files you downloaded and modified to your Databricks cluster
+     ```
+     cd <path-to-init-db-and-install-worker>
+     databricks fs cp init-db.sh dbfs:/spark-dotnet/
+     databricks fs cp install-worker.sh dbfs:/spark-dotnet/
+     ```
+  6. Goto to your Databricks cluster homepage -> Clusters (on the left-side menu) -> Create Cluster
```
Comment (imback82) [inline deployment/README.md]: Go to?

File: deployment/README.md [hunk 3]
```diff
@@ -39,7 +41,7 @@ Microsoft.Spark.Worker is a backend component that lives on the individual worke
    * For example, if you want `.NET for Apache Spark v0.1.0` using `netcoreapp2.1`, you'd download [Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.1.0.tar.gz](https://github.com/dotnet/spark/releases/download/v0.1.0/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.1.0.tar.gz).
 2. Upload `Microsoft.Spark.Worker.<release>.tar.gz` and [install-worker.sh](install-worker.sh) to a distributed file system (e.g., HDFS, WASB, ADLS, S3) that your cluster has access to.
 
-## Your Spark .NET `app`
+## Preparing your Spark .NET App
```
Comment (imback82) [inline deployment/README.md]: I guess Spark .NET here is fine?

File: deployment/README.md [hunk 4]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
-   set +e
+### Run your app on the cloud!
 
-   /bin/bash /dbfs/<your mount>/<path to>/install-worker.sh local /dbfs/<your mount>/<path to>/Microsoft.Spark.Worker.<release>.tar.gz /usr/local/bin
-   """, true)
-   ```
-3. Restart the cluster.
+#### Using [Set JAR](https://docs.databricks.com/user-guide/jobs.html#create-a-job)
+
+> **Note:** This approach allows job submission to an existing active cluster
```
Comment (imback82) [inline deployment/README.md]: end with "."?

File: deployment/README.md [hunk 5]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
+
+One-time Setup:
+  1. Go to your Databricks cluster -> Jobs (on the left-side menu) -> Set JAR
+  2. Upload the appropriate `microsoft-spark-<spark-version>-<spark-dotnet-version>.jar`
+  3. Set the params appropriately:
+     ```
+     Main Class: org.apache.spark.deploy.DotnetRunner
+     Arguments /dbfs/app/<your-app>.zip <your-app-main-class>
```
Comment (imback82) [inline deployment/README.md]: you-app-name?

File: deployment/README.md [hunk 6]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
+     ```
+  4. Configure the Cluster to point to an existing cluster (that already you already set the init script for - see previous section).
```
Comment (imback82) [inline deployment/README.md]: that already you already?

File: deployment/README.md [hunk 7]
```diff
@@ -159,28 +161,59 @@ foo@bar:~$ aws emr add-steps \
...
 
-### Run your app on the cloud!
 #### Using [spark-submit](https://spark.apache.org/docs/latest/submitting-applications.html)
-1. [Create a Job](https://docs.databricks.com/user-guide/jobs.html) and select *Configure spark-submit*.
-2. Configure `spark-submit` with the following parameters:
-   ```shell
-   ["--files","/dbfs/<your mount>/<path-to>/<app assembly/file to deploy to worker>","--class"," org.apache.spark.deploy.DotnetRunner","/dbfs/<your mount>/<path to>/microsoft-spark-<spark_majorversion.spark_minorversion.x>-<spark_dotnet_version>.jar","/dbfs/<your mount>/<path to>/<app name>.zip","<app bin name>","app arg1","app arg2"]
-   ```
+
+> **Note:** This approach allows submission ONLY to cluster that gets created on-demand
```
Comment (imback82) [inline deployment/README.md]: end with "."?

File: deployment/db-init.sh [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+# Change the variables below appropriately. 
+#
+##############################################################################
+################################# CHANGE THESE ###############################
+
+# SPARKDOTNET_RELEASE to point to the appropriate version you downloaded from the
+# https://github.com/dotnet/spark Releases section. For instance, for v0.2.0, you
+# would set it to the following URI:
+# https://github.com/dotnet/spark/releases/download/v0.2.0/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.2.0.tar.gz
+SPARKDOTNET_RELEASE=https://github.com/dotnet/spark/releases/download/v0.2.0/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.2.0.tar.gz
```
Comment (imback82) [inline deployment/db-init.sh]: DOTNET_SPARK?

File: deployment/README.md (hunk 3 above)
Comment (rapoth) [reply deployment/README.md]: Yeah, should be fine.

File: deployment/README.md (hunk 2 above)
Comment (rapoth) [reply deployment/README.md]: Done

File: deployment/README.md (hunk 4 above)
Comment (rapoth) [reply deployment/README.md]: Done

File: deployment/README.md (hunk 5 above)
Comment (rapoth) [reply deployment/README.md]: Done

File: deployment/README.md (hunk 7 above)
Comment (rapoth) [reply deployment/README.md]: Done

File: deployment/db-init.sh (hunk 8 above)
Comment (rapoth) [reply deployment/db-init.sh]: Done

File: deployment/README.md (hunk 6 above)
Comment (rapoth) [reply deployment/README.md]: Done

File: deployment/README.md (hunk 1 above)
Comment (rapoth) [reply deployment/README.md]: Done

**Pull request merged**